sklearn/linear_model/tests/test_perceptron.py

```diff
@@ -40,8 +38,9 @@ def predict(self, X):
         return np.sign(self.project(X))
 
 
-def test_perceptron_accuracy():
-    for data in (X, X_csr):
+@pytest.mark.parametrize("csr_container", CSR_CONTAINERS)
```

Comment:
Instead of running the loop with the dense version of `X` twice, let's parametrize the numpy case as follows:

```suggestion
@pytest.mark.parametrize("data_container", CSR_CONTAINERS + [np.asarray])
```

and then remove the top level for loop in the body of the test.